app/models/branch_meal.rb

```diff
@@ -0,0 +1,26 @@
+class BranchMeal < ApplicationRecord
+  self.table_name = "branches_meals"
+
+  ### ASSOCIATIONS ###
+
+  belongs_to :branch
+  belongs_to :meal
+  has_many :ingredients_meals, through: :meal
+  has_many :inventories, through: :branch
+
+  ### VALIDATIONS ###
+
+  validates :meal_id, uniqueness: { scope: :branch_id }
+
+  ### CALLBACKS ###
+
+  after_save_commit :update_available
```

Comment:
add these callbacks in respective models.

Reply:
Done